crates/stc_ts_file_analyzer/src/analyzer/stmt/var_decl.rs

```diff
@@ -262,33 +262,68 @@ impl Analyzer<'_, '_> {
                         ty.freeze();
                         self.report_error_for_invalid_rvalue(span, &v.name, &ty);
 
-                        self.scope.this = Some(if self.ctx.in_class_member {
-                            if self.scope.this().is_some() {
-                                self.scope.this().unwrap().into_owned()
-                            } else {
-                                Type::This(ThisType {
-                                    span,
-                                    metadata: Default::default(),
-                                    tracker: Default::default(),
-                                })
-                            }
-                        } else if matches!(self.scope.kind(), ScopeKind::ArrowFn) {
-                            if self.scope.this().is_some() {
-                                self.scope.this().unwrap().into_owned()
-                            } else {
+                        self.scope.this = Some(
+                            if matches!(
+                                ty.normalize(),
                                 Type::Query(QueryType {
-                                    span,
-                                    expr: box QueryExpr::TsEntityName(RTsEntityName::Ident(RIdent::new(
-                                        "globalThis".into(),
-                                        span.with_ctxt(SyntaxContext::empty()),
-                                    ))),
-                                    metadata: Default::default(),
-                                    tracker: Default::default(),
+                                    expr: box QueryExpr::TsEntityName(RTsEntityName::TsQualifiedName(box RTsQualifiedName {
```

Comment:
I think this branch is reachable using other types than `globalThis`

Reply:
Please use `ty.is_global_this()` instead.
https://rustdoc.stc.dudy.dev/stc_ts_types/enum.Type.html#method.is_global_this

Reply:
this code mean `typeof this`

Reply:
just contains globalThis, this mean never only globalThis

Reply:
Do symbols also use query types?

Reply:
I'll check it out tomorrow.